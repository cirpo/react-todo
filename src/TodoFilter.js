import React from 'react'
import FlatButton from 'material-ui/FlatButton'
import Badge from 'material-ui/Badge'
import injectTapEventPlugin from 'react-tap-event-plugin'
injectTapEventPlugin()

const style = {padding: '22px 0px 0px 12px'}
const badgeStyle = {top: 5, right: 5}

let TodoFilter = ({todos, filter, onFilter}) => (
  <div>
    <Badge
      badgeContent={todos.length}
      secondary={true}
      badgeStyle={badgeStyle}
      style={style}>
      <FlatButton onClick={() => onFilter(undefined)} label="all" secondary={filter.done === undefined} />
    </Badge>
    <Badge
      badgeContent={todos.filter((todo) => !todo.done).length }
      secondary={true}
      badgeStyle={badgeStyle}
      style={style}>
      <FlatButton onClick={() => onFilter(false)} label="active" secondary={filter.done === false} />
    </Badge>
    <Badge
      badgeContent={todos.filter((todo) => todo.done).length}
      secondary={true}
      badgeStyle={badgeStyle}
      style={style}>
      <FlatButton onClick={() => onFilter(true)} label="done" secondary={filter.done === true} />
    </Badge>
  </div>
)

export default TodoFilter
