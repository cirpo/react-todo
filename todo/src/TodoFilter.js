import React from 'react'

let TodoFilter = ({todos, filter, onFilter}) => (
    <div>
        <span className="mdl-badge mdl-badge--overlap" data-badge={todos.length}
          onClick={() => onFilter(undefined)}>
          <button className={'mdl-button mdl-js-button' + (filter.done === undefined ? ' mdl-button--accent' : '')}>
          ALL
          </button>
        </span>
        <span className="mdl-badge mdl-badge--overlap" data-badge={todos.filter((todo) => !todo.done).length }
          onClick={() => onFilter(false)}>
          <button className={'mdl-button mdl-js-button' + (filter.done === false ? ' mdl-button--accent' : '')}>
          ACTIVE
          </button>
        </span>
        <span className="mdl-badge mdl-badge--overlap" data-badge={todos.filter((todo) => todo.done).length}
          onClick={() => onFilter(true)}>
          <button className={'mdl-button mdl-js-button' + (filter.done === true ? ' mdl-button--accent' : '')}>
          DONE
          </button>
        </span>
    </div>
)

export default TodoFilter
