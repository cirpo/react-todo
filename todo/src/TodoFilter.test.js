import React from 'react'
import TodoFilter from './TodoFilter'
import renderer from 'react-test-renderer'

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <TodoFilter todos={[{"title": 'lol', "done": false}]} filter={{"done": undefined}} />
  );

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  // manually trigger the callback
  tree.props.onMouseEnter()
  // re-rendering
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  // manually trigger the callback
  tree.props.onMouseLeave()
  // re-rendering
  tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
