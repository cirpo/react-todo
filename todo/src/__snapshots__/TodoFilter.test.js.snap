exports[`test Link changes the class when hovered 1`] = `
<div>
  <span
    className="mdl-badge"
    data-badge={1}
    onClick={[Function]}>
    <button
      className="mdl-button mdl-js-button mdl-button--accent">
      ALL
    </button>
  </span>
  <span
    className="mdl-badge"
    data-badge={1}
    onClick={[Function]}>
    <button
      className="mdl-button mdl-js-button mdl-button--accent">
      ACTIVE
    </button>
  </span>
  <span
    className="mdl-badge"
    data-badge={0}
    onClick={[Function]}
    secondary={false}>
    <button
      className="mdl-button mdl-js-button mdl-button--accent">
      DONE
    </button>
  </span>
</div>
`;
